python/python_direct/ops.cpp

```diff
@@ -1253,13 +1301,105 @@ Val or TensorView
 )")
 };
 
+void bindReductionOps(py::module_& ops) {
+  NVFUSER_DIRECT_BINDING_REDUCTION_OP(
+      "max",
+      max,
+      R"(
+Reduce a tensor by computing the maximum value along specified dimensions.
+
+Parameters
+----------
+arg : TensorView
+    Input tensor to reduce.
+dim : int, optional
+    Dimension to reduce over. If not specified, reduces over all dimensions.
+keep_dim : bool, optional
+    Whether to keep the reduced dimensions with size 1. Default is False.
+dtype : PrimDataType, optional
+    Output data type. If None, use the input data type.
+
+Returns
+-------
+TensorView
+    A new tensor containing the maximum values along the specified dimensions.
+)")
+  NVFUSER_DIRECT_BINDING_REDUCTION_OP(
+      "min",
+      min,
+      R"(
+Reduce a tensor by computing the minimum value along specified dimensions.
+
+Parameters
+----------
+arg : TensorView
+    Input tensor to reduce.
+dim : int, optional
+    Dimension to reduce over. If not specified, reduces over all dimensions.
+keep_dim : bool, optional
+    Whether to keep the reduced dimensions with size 1. Default is False.
+dtype : PrimDataType, optional
+    Output data type. If None, use the input data type.
+
+Returns
+-------
+TensorView
+    A new tensor containing the minimum values along the specified dimensions.
+)")
+  NVFUSER_DIRECT_BINDING_REDUCTION_OP(
+      "prod",
+      prod,
+      R"(
+Reduce a tensor by computing the product of elements along specified dimensions.
+
+Parameters
+----------
+arg : TensorView
+    Input tensor to reduce.
+dim : int, optional
+    Dimension to reduce over. If not specified, reduces over all dimensions.
+keep_dim : bool, optional
+    Whether to keep the reduced dimensions with size 1. Default is False.
+dtype : PrimDataType, optional
+    Output data type. If None, use the input data type.
```

Comment:
I think it's worth noting that dtype here is actually used as `math` type, since we cast inputs before calling reduction.
https://github.com/NVIDIA/Fuser/blob/4d361597aab437f0d308a38d0fdb5ef83f88eb55/csrc/ops/arith.cpp#L1519